File: 1291-sequential-digits/1291-sequential-digits.java
class Solution {
    public static List<Integer> sequentialDigits(int low, int high) {
        String start = String.valueOf(low), end = String.valueOf(high);
        String digits = "123456789";

        List<Integer> res = new ArrayList<>();
        Set<Integer> st = new TreeSet<>();

        for (int len = start.length(); len <= end.length(); len++) {
            for (int i = 0; i <= digits.length() - len; i++) {
                String temp = digits.substring(i, i + len);
                st.add(Integer.parseInt(temp));
            }
        }

        for (Integer i : st) {
            if (i >= low && i <= high) {
                res.add(i);
            }
        }

        return res;
    }
}

File: 1291-sequential-digits/1291-sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        string start = to_string(low), end = to_string(high);
        string digits = "123456789";
        
        vector<int>res;
        set<int>st;
        
        for(int len = start.length(); len <= end.length(); len++){
            for(int i = 0; i < digits.length() - len + 1; i++){
                string temp = digits.substr(i, len);
                st.insert(stoi(temp));
            }
        }
        
        for(auto it : st){
            if(it >= low && it <= high){
                res.push_back(it);
            }
        }
        
        return res;
    }
};